app/views/view_data/time_period/edit.html.erb

```diff
@@ -0,0 +1,107 @@
+<main id="content" role="main">
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <a class="link-back" href="<%= @referer %>">Back</a>
+        <% if @settings && @settings.invalid? %>
+          <div class="error-summary" role="alert" aria-labelledby="error-summary-heading-example-1" tabindex="-1">
+            <h2 class="heading-medium error-summary-heading" id="error-summary-heading-example-1">
+               Please fix the following errors:
+            </h2>
+            <ul class="error-summary-list">
+              <% if @settings.errors.any? %>
+                <% @settings.errors.each do |field, errors| %>
+                    <li><a href=<%= "#" + field.to_s %>><%= field.to_s.humanize %> <%= [*errors].join('') %></a></li>
+                  <% end %>
+                <% end %>
+            </ul>
+          </div>
+        <% end %>
+
+      <h1 class="heading-large">What time period do you want to view?</h1>
+
+      <%= form_tag view_data_time_period_path, method: :post, class: "form" do %>
+        <%= hidden_field_tag 'next', @referer %>
+        <fieldset>
+          <legend class="visuallyhidden">
+             <h1 class="heading-medium">Change time period you want to view data for</h1>
+          </legend>
+          <p>
+            You are currently viewing the <%= @time_period.months_covered %> months from
+            <%= time_tag @time_period.starts_on, @time_period.starts_on.to_formatted_s(:abbreviated_day_month_year) %>
+            to
+            <%= time_tag @time_period.ends_on, @time_period.ends_on.to_formatted_s(:abbreviated_day_month_year) %>
+          </p>
+
+          <div class="form-group <%= 'form-group-error' if field_error?(@settings, [:range])%>">
+            <% if field_error?(@settings, [:range]) %>
+              <%= render partial: 'field_message', locals: {settings: @settings, field: :range } %>
+             <% end %>
+            <div class="multiple-choice">
+              <%= radio_button_tag :range, '12'%>
+              <%= label :range, 'Last 12 Months' %>
+            </div>
+            <div class="multiple-choice">
+              <%= radio_button_tag :range, '24' %>
+              <%= label :range, 'Last 24 Months' %>
+            </div>
+            <div class="multiple-choice">
+              <%= radio_button_tag :range, '36' %>
+              <%= label :range, 'Last 36 Months' %>
+            </div>
+          </div>
+        </fieldset>
+
+
+        <div class="form-group">
+          <p class="form-block">or</p>
+        </div>
+
+        <div class="form-group">
+          <div class="multiple-choice">
+          <%= radio_button_tag :range, 'custom', @settings.range == "custom" ? true : false %>
+          <%= label :range, 'Custom date range' %>
+          </div>
+        </div>
+
+        <div id="range" style="<%= 'display:none;' if !field_error?(@settings, [:start_date_month, :start_date_year, :end_date_month, :end_date_year]) %>">
```

Comment:
Do we need to add ```<%= 'form-group-error' if field_error?(@settings, [:range])%>``` to this div as well?  That way if you submit without choosing any range it'll be highlighted too?